applications/asset_tracker_v2/tests/location_module/src/location_module_test.c

```diff
@@ -0,0 +1,563 @@
+	data_module_event->data.cfg.location_timeout = 30;
+	data_module_event->data.cfg.no_data.gnss = false;
+	data_module_event->data.cfg.no_data.neighbor_cell = false;
+
+	ret = LOCATION_MODULE_EVT_HANDLER((struct app_event_header *)data_module_event);
+
+	app_event_manager_free(data_module_event);
+
+	TEST_ASSERT_EQUAL(0, ret);
+}
+
+/* Set location module into state where location_request() has been called.
+ */
+static void setup_location_module_in_active_state(void)
+{
+	/* Pre-condition. */
+	setup_location_module_in_running_state();
+
+	/* Set up expectations.  */
+	/* Set callback to validate location module events. */
+	__wrap__event_submit_Stub(&validate_location_module_evt);
+
+	/* Set default location configuration. */
+	enum location_method methods[] = { LOCATION_METHOD_GNSS, LOCATION_METHOD_CELLULAR };
+
+	struct location_config config_defaults = {
+		.methods_count = 2,
+		.interval = 0,
+		.timeout = 300000,
+		.mode = LOCATION_REQ_MODE_FALLBACK,
+		.methods = {
+			{.method = LOCATION_METHOD_GNSS, .gnss.timeout = 120000},
+			{.method = LOCATION_METHOD_CELLULAR, .cellular.timeout = 30000}
+		}
+	};
+	__wrap_location_config_defaults_set_Expect(NULL, 2, methods);
+	__wrap_location_config_defaults_set_IgnoreArg_config();
+	__wrap_location_config_defaults_set_ReturnThruPtr_config(&config_defaults);
+
+	/* Location configuration modified by location module for location request. */
+	struct location_config config_location_request = {
+		.methods_count = 2,
+		.interval = 0,
+		.timeout = 300000,
+		.mode = LOCATION_REQ_MODE_FALLBACK,
+		.methods = {
+			{.method = LOCATION_METHOD_GNSS, .gnss.timeout = 30000},
+			{.method = LOCATION_METHOD_CELLULAR, .cellular.timeout = 11000}
+		}
+	};
+	__wrap_location_request_ExpectAndReturn(&config_location_request, 0);
+	__wrap_location_request_IgnoreArg_config(); /* TODO: REMOVE */
+
+	/* Send APP_EVT_DATA_GET with APP_DATA_LOCATION. */
+	__wrap_app_event_manager_alloc_ExpectAnyArgsAndReturn(&app_module_event_memory);
+	__wrap_app_event_manager_free_ExpectAnyArgs();
+	struct app_module_event *app_module_event = new_app_module_event();
+
+	app_module_event->type = APP_EVT_DATA_GET;
+	app_module_event->count = 1;
+	app_module_event->data_list[0] = APP_DATA_LOCATION;
+
+	__wrap_app_event_manager_alloc_IgnoreAndReturn(&location_module_event_memory);
+	bool ret = LOCATION_MODULE_EVT_HANDLER((struct app_event_header *)app_module_event);
+
+	app_event_manager_free(app_module_event);
+
+	TEST_ASSERT_EQUAL(0, ret);
+}
+
+/* Test whether the location module generates an event
+ * - for A-GPS data on receiving an A-GPS data request from the location library.
+ * - the location module generates an event with GNSS position on receiving a
+ *   location from the location library.
+ */
+void test_location_gnss_with_agps(void)
+{
+	/* Set expected location module events. */
+	expected_location_module_event_count = 1;
+	expected_location_module_events[0].type = LOCATION_MODULE_EVT_ACTIVE;
+
+#if 0
```

Comment:
These tests started to fail only on qemu_cortex_m3 for an unknown reason (around the time previous upmerge was merged) and I haven't been able to reproduce it locally so I'm trying to figure out what I need to do to make it pass.

Commit "location_module_test trials" should be removed at some point when I know what to fix.

Reply:
It looks like CONFIG_MAIN_STACK_SIZE=4096 might help for this. Tests are running now.

Reply:
Yep, test issues resolved with stack size increase.